miner/src/block_assembler.rs

```diff
@@ -614,28 +614,10 @@ mod tests {
             .parent_hash(parent_hash);
 
         let block = BlockBuilder::default()
-            .uncles(
-                uncles
-                    .into_iter()
-                    .map(TryInto::try_into)
-                    .collect::<Result<_, _>>()
-                    .unwrap(),
-            )
-            .transaction(cellbase.try_into().unwrap())
-            .transactions(
-                transactions
-                    .into_iter()
-                    .map(TryInto::try_into)
-                    .collect::<Result<_, _>>()
-                    .unwrap(),
-            )
-            .proposals(
-                proposals
-                    .into_iter()
-                    .map(TryInto::try_into)
-                    .collect::<Result<_, _>>()
-                    .unwrap(),
-            )
+            .uncles(uncles.into_iter().map(Into::into).collect())
```

Comment:
(not a request change, just a comment)

```rust
impl BlockBuilder {
    fn uncles<T>(mut self, arg: Vec<T>) -> Self where T: Into<UncleBlock>;
    fn uncle<T>(mut self, arg: T) -> Self where T: Into<UncleBlock>;
}
```

Reply:
I prefer (like std does):
``` rust
impl BlockBuilder {
    fn uncles<T: Into<UncleBlock>>(mut self, arg: Vec<T>) -> Self;
    fn uncle<T: Into<UncleBlock>>(mut self, arg: T) -> Self;
}
```

Reply:
> I prefer (like std does):
> 
> ... ...

How many or how long you will change to the `where` statement?

Try to run `rustfmt` for this:
```rust
impl BlockBuilder {
    fn new<H: Into<Header>, U: Into<UncleBlock>, T: Into<Transaction>, O: Into<Something>>(mut self, header: H, uncles: Vec<U>, transactions: Vec<T>, others: O) -> Self {}
}
```

So, why do not use a consistent syntax?

Reply:
update